VultisigApp/VultisigApp/Services/KyberSwap/KyberSwapService.swift

```diff
@@ -0,0 +1,313 @@
+        if let errorResponse = try? JSONDecoder().decode(KyberSwapErrorResponse.self, from: routeData) {
+            if errorResponse.code != 0 {
+                print("🚨 KyberSwap Route Error: \(errorResponse)")
+                throw KyberSwapError.apiError(code: errorResponse.code, message: errorResponse.message, details: errorResponse.details)
+            }
+        }
+        
+        let routeResponse = try JSONDecoder().decode(KyberSwapRouteResponse.self, from: routeData)
+        
+        // Now build the transaction
+        let buildUrl = Endpoint.buildKyberSwapTransaction(chain: chain)
+        
+        let buildPayload = KyberSwapBuildRequest(
+            routeSummary: routeResponse.data.routeSummary,
+            sender: from,
+            recipient: from,
+            slippageTolerance: 100, // 1.0% to match route request
+            deadline: Int(Date().timeIntervalSince1970) + 1200 // 20 minutes from now
+        )
+        
+        var buildRequest = URLRequest(url: buildUrl)
+        buildRequest.httpMethod = "POST"
+        buildRequest.allHTTPHeaderFields = [
+            "accept": "application/json",
+            "content-type": "application/json"
+        ]
+        buildRequest.httpBody = try JSONEncoder().encode(buildPayload)
+        
+        let (buildData, _) = try await URLSession.shared.data(for: buildRequest)
+        
+        // Check for API errors in build response
+        if let errorResponse = try? JSONDecoder().decode(KyberSwapErrorResponse.self, from: buildData) {
+            if errorResponse.code != 0 {
+                print("🚨 KyberSwap Build Error: \(errorResponse)")
+                throw KyberSwapError.apiError(code: errorResponse.code, message: errorResponse.message, details: errorResponse.details)
+            }
+        }
+        
+        var buildResponse = try JSONDecoder().decode(KyberSwapQuote.self, from: buildData)
+        
+        // Add gas price from route response to the build response
+        let gasPrice = routeResponse.data.routeSummary.gasPrice
+        buildResponse.data.gasPrice = gasPrice
+        
+        // Calculate fee using the SAME gas source as the transaction construction
+        // Use build response gas (more accurate) instead of route response gas
+        let baseGas = BigInt(buildResponse.data.gas) ?? BigInt.zero
+        
+        // Add debug logging to diagnose zero fee issue
+        print("🔍 KyberSwap Fee Debug:")
+        print("   Route Gas: '\(routeResponse.data.routeSummary.gas)'")
+        print("   Build Gas: '\(buildResponse.data.gas)'")
+        print("   GasPrice from API: '\(gasPrice)'")
+        print("   BaseGas parsed: \(baseGas)")
+        
+        // Chain-specific gas buffer based on gas costs and execution characteristics
+        // Ethereum: Expensive gas, conservative buffer
+        // L2s: Cheap gas, can afford larger buffer for complex routing
+        let gasMultiplierTimes10: Int
```

Comment:
These logic are the same as https://github.com/vultisig/vultisig-ios/pull/2360/files#diff-1bad0b98b745af44869eeb8d7923215cf83ddb9010897ca9fc182b9dd2fa05d5R100-R116

Can we make sure these logic is in one place?